dot/core/service_test.go

```diff
@@ -28,6 +36,78 @@ import (
 
 var errTestDummyError = errors.New("test dummy error")
 
+func generateTestCentrifugeMetadata(t *testing.T) *ctypes.Metadata {
+	t.Helper()
+	rawMeta := common.MustHexToBytes(testdata.NewTestMetadata())
+	var decoded []byte
+	err := scale.Unmarshal(rawMeta, &decoded)
+	require.NoError(t, err)
+
+	meta := &ctypes.Metadata{}
+	err = ctypes.DecodeFromBytes(decoded, meta)
+	require.NoError(t, err)
+	return meta
+}
+
+func generateExtrinsic(t *testing.T) (ext types.Extrinsic, externExt types.Extrinsic, body *types.Body) {
```

Comment:
nit

```suggestion
func generateExtrinsic(t *testing.T) (ext, externExt types.Extrinsic, body *types.Body) {
```

Also imo `extrinsic` > `ext` and `externalExtrinsic` > `externExt` :wink: